Specialization/Fury.lua

```diff
@@ -1,164 +1,223 @@
+        return FR.ThunderousRoar
     end
 
-    if canExecute then
-        if talents[FR.Massacre] and cooldown[FR.ExecuteMassacre].ready then
-            return FR.ExecuteMassacre;
-        end
-        if not talents[FR.Massacre] and cooldown[FR.Execute].ready then
-            return FR.Execute;
-        end
+    -- odyns_fury,if=buff.enrage.up&(spell_targets.whirlwind>1|raid_event.adds.in>15)
+    if talents[FR.OdynsFury] and cooldown[FR.OdynsFury].ready and (buff[FR.Enrage].up and targets > 1) then
+        return FR.OdynsFury
     end
 
-    if enraged then
-        --if talents[FR.SpearOfBastion] and cooldown[FR.SpearOfBastion].ready then
-        --    return FR.SpearOfBastion;
-        --end
-        if talents[FR.Ravager] and cooldown[FR.Ravager].ready then
-            return FR.Ravager;
-        end
-        if talents[FR.ThunderousRoar] and cooldown[FR.ThunderousRoar].ready then
-            return FR.ThunderousRoar;
-        end
-        if talents[FR.OdynsFury] and cooldown[FR.OdynsFury].ready then
-            return FR.OdynsFury;
-        end
-        if talents[FR.Onslaught] and cooldown[FR.Onslaught].ready then
-            return FR.Onslaught;
-        end
+    -- onslaught,if=!talent.annihilator&buff.enrage.up|talent.tenderize
+    if talents[FR.Onslaught] and cooldown[FR.Onslaught].ready and (not talents[FR.Annihilator] and buff[FR.Enrage].up or talents[FR.Tenderize]) then
+        return FR.Onslaught
     end
 
-    if not talents[FR.Annihilator] and cooldown[FR.RagingBlow].charges >= 1 then
-        return FR.RagingBlow;
+    -- raging_blow,if=charges>1
+    if not recklessnessEmpowered and talents[FR.RagingBlow] and cooldown[FR.RagingBlow].ready and (cooldown[FR.RagingBlow].charges > 1) then
+        return FR.RagingBlow
     end
-	
-	if not talents[FR.TitanicRage] and cooldown[FR.RagingBlowEmpowered].charges >= 1 then
-        return FR.RagingBlowEmpowered;
+
+    -- crushing_blow,if=charges>1
+    if recklessnessEmpowered and talents[FR.RagingBlow] and cooldown[FR.CrushingBlow].ready and (cooldown[FR.CrushingBlow].charges > 1) then
+        return FR.CrushingBlow
     end
 
-    if cooldown[FR.Bloodthirst].ready then
-        return FR.Bloodthirst;
+    -- bloodbath,if=buff.enrage.down|talent.annihilator
+    if recklessnessEmpowered and talents[FR.Bloodthirst] and cooldown[FR.Bloodbath].ready and (not buff[FR.Enrage].up or talents[FR.Annihilator]) then
+        return FR.Bloodbath
     end
 
-    if targets > 1 then
-        return FR.Whirlwind
+    -- bloodthirst,if=talent.annihilator
+    if not recklessnessEmpowered and talents[FR.Bloodthirst] and cooldown[FR.Bloodthirst].ready and (talents[FR.Annihilator]) then
+        return FR.Bloodthirst
     end
 
-    return FR.Slam;
+    -- rampage
+    if talents[FR.Rampage] and rage >= 80 then
+        return FR.Rampage
+    end
+
+    -- slam,if=talent.annihilator
+    if rage >= 20 and (talents[FR.Annihilator]) then
```

Comment:
I find casting slam in multi target situations to be a bit suspect. (even with annihilator). I'm going to look into it a bit.

Reply:
With "Improved Whirlwind" every single target spell also affects 2-4 other targets. So this alone is not a counter argument against using slam ;)

Reply:
ahh, yes. Thank you. That clears up my confusion.